applications/rag/README.md

```diff
@@ -1,192 +1,204 @@
+**NOTE:** Terraform keeps state metadata in a local file called `terraform.tfstate`. Deleting the file may cause some resources to not be cleaned up correctly even if you delete the cluster. We suggest using `terraform destroy` before reapplying/reinstalling.
 
 1. `cd ai-on-gke/applications/rag`
 
-2. Edit `workloads.tfvars` to set your project ID, location, cluster name, and GCS bucket name. Optionally, make the following changes:
-    * Set `create_cluster = false` if you are using an existing cluster.
-     * (Recommended) Set `jupyter_add_auth = true`, `frontend_add_auth = true` and `ray_dashboard_add_auth = true` to create load balancers with IAP for your Jupyter notebook, TGI frontend and Ray Dashboard.
-    * (Recommended) Set `jupyter_domain`, `frontend_domain`, `ray_dashboard_domain` to specify a domain, you need to be able to configure DNS records of that domain, you can [register a Domain on Google Cloud Domains](https://cloud.google.com/domains/docs/register-domain#registering-a-domain), or use the domain registrar of your choice.
-    * Choose a custom k8s namespace and service account to be used by the application.
-    * Set `create_network = true` if you want to create a new VPC network
+2. Edit `workloads.tfvars` to set your project ID, location, cluster name, and GCS bucket name. Ensure the `gcs_bucket` name is unique across all GCP projects (add a random suffix). Optionally, make the following changes:
+    * (Optional) Set a custom `kubernetes_namespace` where all k8s resources will be created.
+    * (Recommended) Enable authenticated access for JupyterHub, frontend chat and Ray dashboard services:
+        - First, make sure the [OAuth Consent Screen](https://developers.google.com/workspace/guides/configure-oauth-consent#configure_oauth_consent) is configured for your project. Ensure `User type` is set to `Internal`.
+        - Next, set the following variables in `workloads.tfvars`:
+            - `jupyter_add_auth = true`
+            - `frontend_add_auth = true`
+            - `ray_dashboard_add_auth = true`
+        - Configure custom domains names via `jupyter_domain`, `frontend_domain` and `ray_dashboard_domain` for your services. If left blank, we'll provision test `nip.io` domains for you.
+        - Configure DNS records for your custom domains: [Register a Domain on Google Cloud Domains](https://cloud.google.com/domains/docs/register-domain#registering-a-domain), or use a domain registrar of your choice. You can skip this if using `nip.io`.
+        - Allowlist principals for your services via `jupyter_members_allowlist`, `frontend_members_allowlist` and `ray_dashboard_members_allowlist`.
+    * (Not recommended) Set `create_cluster = false` if you are using an existing cluster. If using a GKE Standard cluster, ensure it has an L4 nodepool with autoscaling and node autoprovisioning enabled.
```

Comment:
Why mention this here if it's not recommended?

Reply:
because we want to still support existing clusters, even though it's not recommended